pkg/api/engine.go

```diff
@@ -59,7 +59,7 @@ type Engine interface {
 	QueueRun(request *RunRequest, sources *UnpackedSources) (string, error)
 
 	DoBuildPurge(ctx context.Context, builder, plan string, ow *rpc.OutputWriter) error
-	DoCollectOutputs(ctx context.Context, runner string, runID string, ow *rpc.OutputWriter) error
```

Comment:
`DoCollectOutputs` now retrieves a `Task` based on the `runID`, and depending on where this task was executed, fetched the outputs. This makes it possible for the dashboard to list all tasks (no matter which runner was used to run them), and provide links to their outputs.